src/QsCompiler/SyntaxProcessor/ScopeContext.fs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.SyntaxProcessing
+
+open Microsoft.Quantum.QsCompiler
+open Microsoft.Quantum.QsCompiler.SymbolManagement
+open Microsoft.Quantum.QsCompiler.SyntaxProcessing.TypeInference
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open System
+
+/// The context used for symbol resolution and type checking within the scope of a callable.
+type ScopeContext =
+    {
+        /// The namespace manager for global symbols.
+        Globals: NamespaceManager
+
+        /// The symbol tracker for the parent callable.
+        Symbols: SymbolTracker
+
+        /// The type inference context.
+        Inference: InferenceContext
+
+        /// True if the parent callable for the current scope is an operation.
+        IsInOperation: bool
+
+        /// The return type of the parent callable for the current scope.
+        ReturnType: ResolvedType
+
+        /// The runtime capability of the compilation unit.
+        Capability: RuntimeCapability
+
+        /// The name of the processor architecture for the compilation unit.
+        ProcessorArchitecture: string
```

Comment:
I know this was there in the previous version and you are just preserving it, but why do we need the processor architecture for syntax processing?

Reply:
It's used only for diagnostic messages, so they can include the name of the quantum processor in capability errors.

Reply:
Ah, ok. That does make sense. I have been so caught up in working on clang, LLVM, and CMakeLists.txt setup that I misinterpreted this as the classical processor architecture that the code was compiled on, which for Q# didn't seem relevant at all. Thanks for explaining!